src/simtools/applications/derive_psf_parameters.py

```diff
@@ -411,34 +172,8 @@ def main():  # noqa: D103
         model_version=args_dict["model_version"],
     )
 
-    all_parameters = []
-    mrra_0, mfr_0, mrra2_0, mar_0 = get_previous_values(tel_model, logger)
-
-    n_runs = 5 if args_dict["test"] else 50
-    generate_random_parameters(
-        all_parameters, n_runs, args_dict, mrra_0, mfr_0, mrra2_0, mar_0, logger
-    )
-
-    data_to_plot, radius = load_and_process_data(args_dict)
-
-    # Preparing figure name
-    plot_file_name = "_".join((label, tel_model.name + ".pdf"))
-    plot_file = output_dir.joinpath(plot_file_name)
-    pdf_pages = PdfPages(plot_file)
-
-    best_pars, _ = find_best_parameters(
-        all_parameters, tel_model, site_model, args_dict, data_to_plot, radius, pdf_pages
-    )
-
-    # Rerunning and plotting the best pars
-    run_pars(tel_model, site_model, args_dict, best_pars, data_to_plot, radius, pdf_pages)
-    plt.close()
-    pdf_pages.close()
-
-    # Printing the results
-    print("Best parameters:")
-    for par, value in best_pars.items():
-        print(f"{par} = {value}")
+    # Run the complete PSF optimization workflow
+    psf_opt.run_psf_optimization_workflow(tel_model, site_model, args_dict, output_dir, logger)
```

Comment:
Do we need to pass on 'logger'? We don't do this anywhere else and each module has its own logger object. I suggest to be consistent with the other code.